xdsl/dialects/csl/csl_stencil.py

```diff
@@ -362,9 +360,8 @@ def verify_(self) -> None:
                 )
 
         # typecheck required (only) block arguments
-        assert isattr(
+        assert (AnyTensorTypeConstr | MemRefType.constr()).matches(
```

Comment:
There are a few of these, and they are all ugly, I guess this is why we had the `TensorOrMemRefOf`. It should get a little prettier with `TensorType.constr()` when we add one in the future.

Reply:
We could have `AnyTensorOrMemRef` without it being its own custom constraint